service/parserprovider/expand.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parserprovider
+
+import (
+	"context"
+	"os"
+
+	"go.opentelemetry.io/collector/config"
+)
+
+type expandMapProvider struct {
+	base MapProvider
+}
+
+// NewExpandMapProvider returns a MapProvider, that expands all environment variables for a
+// config.Map provided by the given MapProvider.
+func NewExpandMapProvider(base MapProvider) MapProvider {
```

Comment:
Suggestion `NewEnvVarExpandMapProvider`, it is a big name but it makes clear what it expands.

Reply:
I think this should be merged with the config source manager later. That's why I kept a different name.